lib/decoder.ts

```diff
@@ -0,0 +1,498 @@
+    }
+
+    this.width = this.readUInt32LE();
+    this.height = this.readUInt32LE();
+
+    this.planes = this.buffer.readUInt16LE(this.pos);
+    this.pos += 2;
+    this.bitPP = this.buffer.readUInt16LE(this.pos);
+    this.pos += 2;
+
+    this.compression = this.readUInt32LE();
+    this.rawSize = this.readUInt32LE();
+    this.hr = this.readUInt32LE();
+    this.vr = this.readUInt32LE();
+    this.colors = this.readUInt32LE();
+    this.importantColors = this.readUInt32LE();
+
+    // De facto defaults
+
+    if (this.bitPP === 32) {
+      this.maskAlpha = 0;
+      this.maskRed = 0x00ff0000;
+      this.maskGreen = 0x0000ff00;
+      this.maskBlue = 0x000000ff;
+    } else if (this.bitPP === 16) {
+      this.maskAlpha = 0;
+      this.maskRed = 0x7c00;
+      this.maskGreen = 0x03e0;
+      this.maskBlue = 0x001f;
+    }
+
+    // End of BITMAP_INFO_HEADER
+
+    if (
+      this.headerSize > HeaderTypes.BITMAP_INFO_HEADER ||
+      this.compression === Compression.BI_BIT_FIELDS ||
+      this.compression === Compression.BI_ALPHA_BIT_FIELDS
+    ) {
+      this.maskRed = this.readUInt32LE();
+      this.maskGreen = this.readUInt32LE();
+      this.maskBlue = this.readUInt32LE();
+    }
+
+    // End of BITMAP_V2_INFO_HEADER
+
+    if (
+      this.headerSize > HeaderTypes.BITMAP_V2_INFO_HEADER ||
+      this.compression === Compression.BI_ALPHA_BIT_FIELDS
+    ) {
+      this.maskAlpha = this.readUInt32LE();
+    }
+
+    // End of BITMAP_V3_INFO_HEADER
+
+    if (this.headerSize > HeaderTypes.BITMAP_V3_INFO_HEADER) {
+      this.pos +=
+        HeaderTypes.BITMAP_V4_HEADER - HeaderTypes.BITMAP_V3_INFO_HEADER;
+    }
+
+    // End of BITMAP_V4_HEADER
+
+    if (this.headerSize > HeaderTypes.BITMAP_V4_HEADER) {
+      this.pos += HeaderTypes.BITMAP_V5_HEADER - HeaderTypes.BITMAP_V4_HEADER;
+    }
+
+    // End of BITMAP_V5_HEADER
+
+    if (this.bitPP <= 8 || this.colors > 0) {
+      const len = this.colors === 0 ? 1 << this.bitPP : this.colors;
+      this.palette = new Array(len);
+
+      for (let i = 0; i < len; i++) {
+        const blue = this.buffer.readUInt8(this.pos++);
+        const green = this.buffer.readUInt8(this.pos++);
+        const red = this.buffer.readUInt8(this.pos++);
+        const quad = this.buffer.readUInt8(this.pos++);
+
+        this.palette[i] = {
+          red,
+          green,
+          blue,
+          quad
+        };
+      }
+    }
+
+    // End of color table
+
+    // Can the height ever be negative?
+    if (this.height < 0) {
+      this.height *= -1;
+      this.bottomUp = false;
+    }
+
+    const coloShift = maskColor(
+      this.maskRed,
+      this.maskGreen,
+      this.maskBlue,
+      this.maskAlpha
+    );
+
+    this.shiftRed = coloShift.shiftRed;
+    this.shiftGreen = coloShift.shiftGreen;
+    this.shiftBlue = coloShift.shiftBlue;
+    this.shiftAlpha = coloShift.shiftAlpha;
+  }
+
+  private parseRGBA() {
+    this.data = Buffer.alloc(this.width * this.height * 4);
+
```

Comment:
A nice additional feature to support is blank space at the end of the header. You can achieve this by using the offset that you parsed earlier and moving pos to that location before extracting pixel data.

```suggestion

    this.pos = this.offset;
```